js/civicrm_stripe.js

```diff
@@ -151,14 +151,12 @@
       // Hide payment if total is 0 and no more participants.
       if ($('#priceset').length) {
         additionalParticipants = cj("#additional_participants").val();
-        // The currentTotal is already being calculated in Form/Contribution/Main.tpl.
-        if(typeof currentTotal !== 'undefined') {
-          if (currentTotal == 0 && !additionalParticipants) {
-            // This is also hit when "Going back", but we already have stripe_token.
-            debugging('ozlkf');
-            // This should not happen on Confirm Contribution, but seems to on 4.6 for some reason.
-            //return true;
-          }
+        currentTotal = $('#pricevalue').data('raw-total');
+        if (currentTotal == 0 && !additionalParticipants) {
+          // This is also hit when "Going back", but we already have stripe_token.
+          debugging('ozlkf');
+          // This should not happen on Confirm Contribution, but seems to on 4.6 for some reason.
+          //return true;
         }
```

Comment:
If currentTotaal == 0 and there are no additional participants, shouldn't we return true? I'm not sure why that is commented out in the original or in your fix.

Reply:
@h-c-c maybe it's better to have separate branches for 4.6 and 4.7? This issue still exists and there doesn't seem to be any progress on testing it on 4.6.

btw, this commit https://github.com/progressivetech/com.drastikbydesign.stripe/commit/b2556639b0c3f1f48f85143c83b86bf40209081e is probably the better one to create as a PR.